### PR TITLE
Support summary/full data download for pie chart

### PR DESCRIPTION
**Pull request by zhx828**
There is one concern in this PR. The submenu of the download is not perfectly aligned on the left side. See the animation below. 
Since we do absolute positioning for submenu, then we need to calculate the left position based on the max width of the submenu item. Personally, I'm not a fan of doing that because you need to update the function when the style of the dropdown updated, for instance, padding/margin/border width. And also when I hover over the download option, my attention has been brought to the submenu, then the key here is actually how to quickly go to the submenu. we also need to consider if we calculate the width, then if the submenu is really wide, due to the items in the submenu are left aligned, then user has to move the mouse quite far away in order to download the data. But please let me know if you think I should try to align the submenu. @jjgao @schultzn @alisman 

Jul-16-2019 15-25-39

# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/6357

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -2782,6 +2783,25 @@ export class StudyViewPageStore {
         return survivalTypes;
     }
 
+    public async getPieChartDataDownload(chartMeta: ChartMeta, dataType?: DataType) {
+        const infoForSummaryDataDownload = this.isCustomChart(chartMeta.uniqueKey) ?
+            this.getCustomChartDataCount(chartMeta) : this.getClinicalDataCount(chartMeta);
+
+        return dataType && dataType === 'summary' ?
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: i think we should avoid nested ternary.  just too hard to read

File: src/pages/studyView/StudyViewUtils.tsx [hunk 2]
```diff
@@ -1238,15 +1238,19 @@ export function pickClinicalAttrFixedColors(data: ClinicalDataCount[]): {[attrib
...
 
-export type ClinicalDataCountWithColor = ClinicalDataCount & { color: string }
+export type ClinicalDataCountSummary = ClinicalDataCount & { color: string , percentage: number, freq: string}
 
-export function getClinicalDataCountWithColorByClinicalDataCount(counts:ClinicalDataCount[]):ClinicalDataCountWithColor[] {
+export function getClinicalDataCountWithColorByClinicalDataCount(counts:ClinicalDataCount[]):ClinicalDataCountSummary[] {
     counts.sort(clinicalDataCountComparator);
     const colors = pickClinicalDataColors(counts);
-    return counts.map(slice =>{
+    const sum = _.sum(counts.map(count => count.count));
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: _.sumBy

File: src/pages/studyView/tabs/SummaryTab.tsx [hunk 3]
```diff
@@ -149,8 +150,8 @@ export class StudySummaryTab extends React.Component<IStudySummaryTabProps, {}>
                     props.onResetSelection = this.handlers.onValueSelection;
                 }
                 props.onChangeChartType = this.handlers.onChangeChartType;
-                props.getData = () => this.store.isCustomChart(chartMeta.uniqueKey) ? this.store.getCustomChartDownloadData(chartMeta) : this.store.getClinicalData(chartMeta);
-                props.downloadTypes = ["Data", "SVG", "PDF"];
+                props.getData = (dataType?: DataType) => this.store.getPieChartDataDownload(chartMeta, dataType);
+                props.downloadTypes = this.store.isCustomChart(chartMeta.uniqueKey) ? ["Summary Data", "SVG", "PDF"] : ["Summary Data", "Full Data", "SVG", "PDF"];
```
Comment (alisman) [inline src/pages/studyView/tabs/SummaryTab.tsx]: maybe a commend explaining why we don't do full data for custom?

Comment (alisman) [reply src/pages/studyView/tabs/SummaryTab.tsx]: this comment might belong somewhere else.  but would be nice to have somewhere

**Comment by alisman** [comment]
@zhx828 does this need particular backend?

**Comment by jjgao** [comment]
The arrow pointing left seems a little strange to me, but maybe it's just me... I also don't have a good solution for this. @alisman what do you think?

**Comment by zhx828** [comment]
@alisman it does not.

File: src/pages/studyView/tabs/SummaryTab.tsx (hunk 3 above)
Comment (zhx828) [reply src/pages/studyView/tabs/SummaryTab.tsx]: I agree. Actually after second thought, the custom chart should support  both downloads. Just updated the pr.

**Comment by jjgao** [comment]
@zhx828 could you make the arror alway point to the right. I think that's more intuitive.

image

**Comment by zhx828** [comment]
@jjgao updated

**Pull request merged**